### PR TITLE
Specifying about representative_datasets in TfLiteConverter

### LINKED ISSUE #37111: representative_data_gen input order?
In tflite post integer quantization, `converter.representative_dataset` is necessary.

However, the documentation never specifies the order of the fed inputs. Are they ordered by lexicographic order of the names, size of shapes or even random order? When multiple inputs are present, it is totally a guessing game.

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #37111 . @mihaimaruseac , Please review.

### CODE REVIEW

**Review by mihaimaruseac — COMMENTED** [review]
Let's also implement the ordering of inputs in the same PR

**Comment by tigert1998** [comment]
It's probably not fixed.

**Comment by ashutosh1919** [comment]
> It's probably not fixed.

@tigert1998 , Please suggest the required changes that you want me to change.

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/lite/python/lite.py
Comment (mihaimaruseac) [inline tensorflow/lite/python/lite.py]: From internal review (CC @shuharsh)

> This is not true, it is optional. it is only necessary if INT8 is the only support builtin ops in target ops.

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/lite/python/lite.py
Comment (mihaimaruseac) [reply tensorflow/lite/python/lite.py]: CC @suharshs

File: tensorflow/lite/python/lite.py
Comment (suharshs) [inline tensorflow/lite/python/lite.py]: This isn't completely true, this only opts you into int8 post-training quantization. Int8 quantization of weights should still happen without this flag.

**Comment by gbaned** [comment]
@ashutosh1919 Can you please check reviewer comments and keep us posted. Thanks!

**Comment by ashutosh1919** [comment]
@mihaimaruseac and @suharshs , I have changed the description as per your comments. Please review it.

**Review by mihaimaruseac — COMMENTED** [review]
LGTM to me but waiting for approval from @suharshs

**Comment by ashutosh1919** [comment]
@suharshs , Please review this PR.

**Review by suharshs — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Pull request merged**
